src/main/java/org/folio/innreach/batch/contribution/service/InitialContributionEventProcessor.java

```diff
@@ -0,0 +1,178 @@
+import static org.folio.innreach.domain.entity.JobExecutionStatus.Status.PROCESSED;
+import static org.folio.innreach.domain.entity.JobExecutionStatus.Status.READY;
+import static org.folio.innreach.domain.entity.JobExecutionStatus.Status.RETRY;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class InitialContributionEventProcessor {
+  @Qualifier("instanceExceptionListener")
+  private final ContributionExceptionListener instanceExceptionListener;
+  @Qualifier("itemExceptionListener")
+  private final ContributionExceptionListener itemExceptionListener;
+  private final InventoryViewService inventoryViewService;
+  private final RecordContributionService recordContributionService;
+  private final ContributionValidationService validationService;
+  private final ContributionJobProperties jobProperties;
+  private final JobExecutionStatusRepository jobExecutionStatusRepository;
+  private static final ConcurrentHashMap<UUID, Contribution> contributionRecord = new ConcurrentHashMap<>();
+  private final ContributionRepository contributionRepository;
+  private final TenantScopedExecutionService executionService;
+  @Value("${initial-contribution.retry-attempts}")
+  private int maxRetryAttempts;
+
+  @Transactional
+  @Async
+  public void processInitialContributionEvents(JobExecutionStatus job) {
+    executionService.runTenantScoped(job.getTenant(), () -> {
+      log.info("processInitialContributionEvents:: Processing Initial contribution events {} ", job);
+      try {
+        var instanceId = job.getInstanceId();
+        var centralServerId = contributionRecord.get(job.getJobId()) != null ?
+          contributionRecord.get(job.getJobId()).getCentralServer().getId() : getCentralServerId(job.getJobId());
+        var instance = inventoryViewService.getInstance(instanceId);
+        if (centralServerId == null || instance == null) {
+          log.warn("processInitialContributionEvents:: Unable to process event with instance " +
+            "id {} centralServerId {} ", instanceId, centralServerId);
+          updateJobAndContributionStatus(job, FAILED, job.isInstanceContributed());
+          return;
+        }
+        checkRetryLimit(job);
+        startContribution(centralServerId, instance, job);
+      } catch (ServiceSuspendedException | InnReachConnectionException | FeignException |
+               SocketTimeOutExceptionWrapper ex) {
+        log.warn("processInitialContributionEvents:: Retrying the contribution for instanceId {} due to {} ",
+          job.getInstanceId(), ex.getMessage());
+        updateJobAndContributionStatus(job, RETRY, job.isInstanceContributed());
+      } catch (Exception ex) {
+        log.warn("processInitialContributionEvents:: Exception while processing instanceId {} ", job.getInstanceId());
+        logException(job, ex, contributionRecord.get(job.getJobId()).getId());
+        updateJobAndContributionStatus(job, FAILED, job.isInstanceContributed());
+      }
+    });
+  }
```

Comment:
Think if we have any ability to add additional details to the log messages here, we should take it before deploying -- speaking from experience trying to debug this in production. For example, I'm not seeing the job id being logged. Basically all the details that can be added should be added.

Reply:
It looks like `JobExecutionStatus` is in scope here and it would have all the details that we might want in each of these log messages for `warn` or `info`. It looks like that is annotated with lombok's `ToString` annotation which should be called by log4j here (at least in the `info` line) into something meaningful. Please LMK if that is correct.

If that's true, then perhaps all of this logging doesn't need to be changed. The instance id is logged in each place and that should be enough to tie it back to the particular `JobExcecutionStatus`. See if you agree.